x-pack/plugin/downsample/src/internalClusterTest/java/org/elasticsearch/xpack/downsample/DownsampleIT.java

```diff
@@ -263,6 +272,53 @@ public void testAggMetricInEsqlTSAfterDownsampling() throws Exception {
                     // TODO: verify the numbers are accurate
                 }
             }
+            // tests on non-downsampled index
+            // TODO: combine with above when support for aggregate_metric_double + implicit casting is added
+            // TODO: add to counter tests below when support for counters is added
+            for (String innerCommand : List.of("first_over_time", "last_over_time")) {
+                String command = outerCommand + " (" + innerCommand + "(cpu))";
+                try (var resp = esqlCommand("TS " + secondIndex + " | STATS " + command + " by cluster, bucket(@timestamp, 1 hour)")) {
+                    var columns = resp.columns();
+                    assertThat(columns, hasSize(3));
+                    assertThat(
+                        "resp is " + resp,
+                        columns,
+                        equalTo(
+                            List.of(
+                                new ColumnInfoImpl(command, expectedType, null),
+                                new ColumnInfoImpl("cluster", "keyword", null),
+                                new ColumnInfoImpl("bucket(@timestamp, 1 hour)", "date", null)
+                            )
+                        )
+                    );
+                }
+            }
+
+            // tests on counter types
+            // TODO: remove hard-coded pragmas
+            var ratePragmas = new QueryPragmas(Settings.builder().put(QueryPragmas.MAX_CONCURRENT_SHARDS_PER_NODE.getKey(), 1).build());
```

Comment:
This test may fail in release builds because pragmas require either a snapshot build or acceptedPragmaRisks to be set to true.